arch/cortex-m/samv71-hal/src/peripherals/mod.rs

```diff
@@ -0,0 +1,41 @@
+//! Access to the hardware peripherals.
+
+pub mod watchdog;
+
+use aerugo_cortex_m::Mutex;
+use samv71q21_pac::CorePeripherals as cortex_peripherals;
+use samv71q21_pac::Peripherals as samv71_peripherals;
+
+/// Mutex indicating whether the peripherals instance have already been taken.
+static PERIPHERALS_TAKEN: Mutex<bool> = Mutex::new(false);
+
+/// Peripherals structure.
+pub struct Peripherals {
+    /// Raw MCU peripherals
+    pub mcu_pac: samv71_peripherals,
+    /// Raw Cortex-M peripherals
+    pub cortex_pac: cortex_peripherals,
+}
+
+impl Peripherals {
+    /// Create new peripherals instance. This function can be called successfully only once.
+    pub fn new() -> Option<Peripherals> {
+        PERIPHERALS_TAKEN.lock(|peripherals_taken| {
+            if *peripherals_taken {
+                return None;
+            }
+
+            *peripherals_taken = true;
```

Comment:
Lock should end here.

Reply:
Yeah, but that would result in uglier code, as i'd have to manage the early return manually instead of relying on return from the lock. Would it even make any real difference? I'm tempted to check the assembly output...

Reply:
Won't make any real difference, it's just a logical thing. This is called in the intitialization code so no interrupt should happen anyway. Leave as is I guess.

Reply:
That will be killed in next MR anyway